drivers/gpio/gpio_brcmstb.c

```diff
@@ -0,0 +1,141 @@
+/*
+ * Copyright (c) 2024 Junho Lee <junho@tsnlab.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT brcm_brcmstb_gpio
+
+#include <zephyr/arch/common/sys_bitops.h>
+#include <zephyr/arch/cpu.h>
+#include <zephyr/device.h>
+#include <zephyr/drivers/gpio.h>
+#include <zephyr/drivers/gpio/gpio_utils.h>
+
+#define GIO_DATA  0x04
+#define GIO_IODIR 0x08
+
+#define DEV_CFG(dev)  ((const struct gpio_brcmstb_config *)(dev)->config)
+#define DEV_DATA(dev) ((struct gpio_brcmstb_data *)(dev)->data)
+
+struct gpio_brcmstb_config {
+	struct gpio_driver_config common;
+
+	DEVICE_MMIO_NAMED_ROM(reg_base);
+	mem_addr_t offset;
+};
+
+struct gpio_brcmstb_data {
+	struct gpio_driver_data common;
+
+	DEVICE_MMIO_NAMED_RAM(reg_base);
+	mem_addr_t base;
+};
+
+static int gpio_brcmstb_pin_configure(const struct device *port, gpio_pin_t pin, gpio_flags_t flags)
+{
+	struct gpio_brcmstb_data *data = port->data;
+
+	if (flags & (GPIO_SINGLE_ENDED | GPIO_PULL_UP | GPIO_PULL_DOWN)) {
+		return -ENOTSUP;
+	}
+
+	if (flags & GPIO_INPUT) {
+		sys_set_bit(data->base + GIO_IODIR, pin);
+	} else if (flags & GPIO_OUTPUT) {
+		sys_clear_bit(data->base + GIO_IODIR, pin);
+
+		if (flags & GPIO_OUTPUT_INIT_HIGH) {
+			sys_set_bit(data->base + GIO_DATA, pin);
+		} else if (flags & GPIO_OUTPUT_INIT_LOW) {
+			sys_clear_bit(data->base + GIO_DATA, pin);
+		}
+	}
+
+	return 0;
+}
+
+static int gpio_brcmstb_port_get_raw(const struct device *port, gpio_port_value_t *value)
+{
+	struct gpio_brcmstb_data *data = port->data;
```

Comment:
Same, please check all pointers for NULL value.

Reply:
the same as above